source/Halibut/Transport/AuthorizedTcpConnectionsLimiter.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright 2012-2013 Octopus Deploy Pty. Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Halibut.Diagnostics;
+using Halibut.Exceptions;
+using Halibut.Util;
+
+namespace Halibut.Transport
+{
+    public interface IAuthorizedTcpConnectionsLimiter
+    {
+        IDisposable ClaimAuthorizedTcpConnection(string thumbprint);
+    }
+
+    public class AuthorizedTcpConnectionsLimiter : IAuthorizedTcpConnectionsLimiter
```

Comment:
don't forget to test this class in isolation, you could smash it with many threads to see how it behaves

Reply:
yep! Working on that as well :)

Reply:
Added in [15e1115](https://github.com/OctopusDeploy/Halibut/pull/599/commits/15e111565121d8cb656742537ffddef1c80752ba)